packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -0,0 +1,123 @@
+  int flex = 1;
+  double extent = 0.0;
+}
+
+/// A sliver that places multiple sliver children in a linear array along the cross
+/// axis.
+/// 
+/// Since the extent of the viewport in the cross axis direction is finite,
+/// each sliver is forced to divide up the extent per specification via
+/// ParentData widgets SliverCrossAxisConstrained and SliverExpanded.
+/// 
+/// The algorithm for dividing up the cross axis extent is as follows.
+/// Every widget either has constrained cross axis extent or a flex value.
+/// First allocate space to all the widgets with constrained cross axis extent.
+/// Then divide up the remaining space proportional to the widget's flex value.
+/// Flex values can be specified via the SliverExpanded widgets, but if
+/// a sliver without SliverExpanded or SliverCrossAxisConstrained is passed,
+/// it is assumed to have a flex value of 1, which by itself simply expands
+/// to take up all of the cross axis extent.
+class RenderSliverCrossAxisGroup extends RenderSliver with ContainerRenderObjectMixin<RenderSliver, SliverFlexParentData> {
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! SliverFlexParentData) {
+      child.parentData = SliverFlexParentData();
+    }
+  }
+
+  @override
+  void performLayout() {
+    // Iterate through each sliver.
+    // Get the parent's dimensions.
+    final double crossAxisExtent = constraints.crossAxisExtent;
+    assert(crossAxisExtent.isFinite);
+
+    // Iterate through each child.
+    int totalFlex = 0;
+    double usedUpExtent = 0.0;
+    RenderSliver? child = firstChild;
+    while (child != null) {
+      final SliverFlexParentData childParentData = child.parentData! as SliverFlexParentData;
+      if (childParentData.extent == 0.0) {
+        totalFlex += childParentData.flex;
+      } else {
+        usedUpExtent += childParentData.extent;
+      }
+      child = childAfter(child);
+    }
+    final double extentPerFlexValue = (crossAxisExtent - usedUpExtent) / totalFlex;
+
+    child = firstChild;
+    double leftSideOffset = 0.0;
+    while (child != null) {
+      final SliverFlexParentData childParentData = child.parentData! as SliverFlexParentData;
+      double childExtent;
+      if (childParentData.extent != 0.0) {
+        childExtent = childParentData.extent;
+      } else {
+        childExtent = extentPerFlexValue * childParentData.flex;
+      }
+      child.layout(constraints.copyWith(
+        crossAxisExtent: childExtent,
+      ), parentUsesSize: true);
+      childParentData.paintOffset = Offset(leftSideOffset, 0.0);
+      leftSideOffset += childExtent;
+      if (geometry == null) {
+        geometry = child.geometry;
+      }
+      else if (geometry!.scrollExtent < child.geometry!.scrollExtent) {
+        geometry = child.geometry;
+      }
+      child = childAfter(child);
+    }
+    geometry ??= SliverGeometry.zero;
```

Comment:
Is there a path where this would still be null?

Reply:
No, but I think the correct way to do this is to set `geometry = SliverGeometry.zero` before looping through the children, in case there's some reason to have no children at all.